AndroidBound/src/main/java/solutions/alterego/androidbound/android/AndroidViewModel.java

```diff
@@ -0,0 +1,104 @@
+package solutions.alterego.androidbound.android;
+
+import android.app.Activity;
+import android.app.FragmentManager;
+import android.os.Bundle;
+
+import java.lang.ref.WeakReference;
+
+import lombok.Getter;
+import lombok.experimental.Accessors;
+import solutions.alterego.androidbound.ViewModel;
+import solutions.alterego.androidbound.android.interfaces.IActivityFocus;
+import solutions.alterego.androidbound.android.interfaces.IActivityLifecycle;
+import solutions.alterego.androidbound.android.interfaces.IHasActivity;
+import solutions.alterego.androidbound.android.interfaces.IHasFragmentManager;
+import solutions.alterego.androidbound.android.interfaces.INeedsActivity;
+import solutions.alterego.androidbound.android.interfaces.INeedsFragmentManager;
+
+@Accessors(prefix = "m")
+public class AndroidViewModel extends ViewModel
+        implements IActivityLifecycle, IActivityFocus, INeedsActivity, IHasActivity, INeedsFragmentManager, IHasFragmentManager {
+
+    private transient WeakReference<Activity> mParentActivity;
+
+    @Getter
+    private boolean mCreated = false;
+
+    @Getter
+    private FragmentManager mFragmentManager;
+
+    @Override
+    public void setParentActivity(WeakReference<Activity> activityRef) {
+        mParentActivity = activityRef;
+    }
+
+    @Override
+    public Activity getParentActivity() {
+        if (mParentActivity != null && mParentActivity.get() != null) {
+            return mParentActivity.get();
+        }
+
+        return null;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        mCreated = true;
+    }
+
+    @Override
+    public void onStart() {
+        //do nothing, to be overridden
+    }
+
+    @Override
+    public void onRestart() {
+        //do nothing, to be overridden
+    }
+
+    @Override
+    public void onStop() {
+        //do nothing, to be overridden
+    }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        //do nothing, to be overridden
+    }
+
+    @Override
+    public void onDestroy() {
+        if (!isDisposed()) {
+            dispose();
+        }
+
+        mCreated = false;
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+
+        if (mParentActivity != null && mParentActivity.get() != null) {
+            mParentActivity.clear();
+        }
+
+        mFragmentManager = null;
+    }
+
+    @Override
+    public void setFragmentManager(FragmentManager fragmentManager) {
+        mFragmentManager = fragmentManager;
+    }
+
+    @Override
+    public void onGotFocus() {
```

Comment:
well, that's why I renamed it - that way people are going to use it only if they have to, and the focus thing is the only reason you would want to use it. otherwise I would rather remove it.

Reply:
also this way you explicitly ask for a certain `FragmentManager` type (regular or support) so you know what you're going to get. Otherwise you would need to check if the manager is the right type but that does you nothing since the Fragments are not compatible between regular and support implementations. This makes things a bit more explicit at the ViewModel level.